machine_test.go

```diff
@@ -252,7 +254,15 @@ func testAttachVsock(ctx context.Context, t *testing.T, m *Machine) {
 func testStartInstance(ctx context.Context, t *testing.T, m *Machine) {
 	err := m.startInstance(ctx)
 	if err != nil {
-		t.Errorf("startInstance failed: %s", err)
+		if syncErr, ok := err.(*operations.CreateSyncActionDefault); ok {
+			if strings.HasPrefix(syncErr.Payload.FaultMessage, "Cannot create vsock device") {
+				t.Errorf(`startInstance: %s
+Do you have permission to interact with /dev/vhost-vsock?
```

Comment:
Given that vsock is disabled in default Firecracker builds, do you think it also makes sense to specifically handle the case where we're running with a build that doesn't support it?

Reply:
Yeah, that's fair.  What would you suggest for that?

Reply:
"Does your installation of Firecracker have support for vsock?  Enable it by building with `cargo build --features vsock`?"

Reply:
Different error, will handle in a different PR:

```
    --- FAIL: TestMicroVMExecution/TestAttachVsock (0.00s)
        machine_test.go:250: attaching vsock failed: [PUT /vsocks/{id}][400] putGuestVsockByIdBadRequest  &{FaultMessage:Invalid request method and/or path: PUT /vsocks/foo}
```

Reply:
Opened https://github.com/firecracker-microvm/firecracker-go-sdk/issues/29 for tracking.